.github/workflows/macos-unit-tests.yml

```diff
@@ -0,0 +1,99 @@
+name: macos-unit-tests
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 8.*
+      - 9.*
+
+permissions:
+  contents: read
+
+## Concurrency only allowed in the main branch.
+## So old builds running for old commits within the same Pull Request are cancelled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  unit-tests-push:
```

Comment:
Can you refactor this using the matrix exclusion instead of using the if condition? See https://docs.github.com/en/actions/how-tos/write-workflows/choose-what-workflows-do/run-job-variations